jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/CommonUtilsTest.java

```diff
@@ -50,6 +56,68 @@
             new TypeAnnotationsScanner(), 
             new FieldAnnotationsScanner(), new SubTypesScanner());
 
+    private List<ProcessDefinition> expected;
+    private List<ProcessDefinition> input;
+
+    @Before
+    public void initComparatorTest(){
+        expected = new ArrayList<ProcessDefinition>() {{
+            add(new ProcessAssetDesc(null, "a1", null, null, null,
+                                     null, null, "org.jbpm.1.service", false));
+            add(new ProcessAssetDesc(null, "A2", null, null, null,
+                                     null, null, "Org.jbpm.2.service", false));
+            add(new ProcessAssetDesc(null, "b1", null, null, null,
+                                     null, null, "ORg.jbpm.3.service", false));
+            add(new ProcessAssetDesc(null, "B2", null, null, null,
+                                     null, null, "ORG.jbpm.4.service", false));
+
+            add(new ProcessAssetDesc(null, "c1", null, null, null,
+                                     null, null, "ORG.Jbpm.5.service", false));
+            add(new ProcessAssetDesc(null, "C2", null, null, null,
+                                     null, null, "ORG.JBpm.6.service", false));
+        }};
+
+        input = new ArrayList<ProcessDefinition>() {{
+
+            add(new ProcessAssetDesc(null, "A2", null, null, null,
+                                     null, null, "Org.jbpm.2.service", false));
+            add(new ProcessAssetDesc(null, "a1", null, null, null,
+                                     null, null, "org.jbpm.1.service", false));
+            add(new ProcessAssetDesc(null, "b1", null, null, null,
+                                     null, null, "ORg.jbpm.3.service", false));
+            add(new ProcessAssetDesc(null, "B2", null, null, null,
+                                     null, null, "ORG.jbpm.4.service", false));
+            add(new ProcessAssetDesc(null, "C2", null, null, null,
+                                     null, null, "ORG.JBpm.6.service", false));
+            add(new ProcessAssetDesc(null, "c1", null, null, null,
+                                     null, null, "ORG.Jbpm.5.service", false));
+        }};
+    }
+
+    @Test
+    public void ignoreCaseComparatorUsingProjectNameTest(){
+
+        initComparatorTest();
+        QueryContext queryContext = new QueryContext();
+        queryContext.setOrderBy("ProcessName");
+        queryContext.setAscending(true);
+        assertNotEquals(expected,input);
+        Collections.sort(input,CommonUtils.getIgnoreCaseComparator(queryContext));
+
+        assertEquals(expected,input);
+    }
+
+    @Test
+    public void ignoreCaseComparatorUsingProjectTest(){
+        initComparatorTest();
+        QueryContext queryContext = new QueryContext();
+        queryContext.setOrderBy("Project");
+        queryContext.setAscending(true);
```

Comment:
Can you also check that `setAscending(false)` also works as expected?

Reply:
if use setAscending(false), it is a other test case.
I will create new one